core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2788,6 +2788,18 @@ public static double log(BigDecimal d0, BigDecimal d1) {
     return Math.log(d0.doubleValue()) / Math.log(d1.doubleValue());
   }
 
+  /** SQL {@code LOG2(number)} function applied to double values. */
+  public static double log2(double d0) {
+    return (Double.isInfinite(log(d0, 2)) && d0 == 0.0) ? Double.NaN : log(d0, 2);
```

Comment:
Because - infinity and + infinity are special cases in SQL and do not support parsing, so I added the condition here that d0 == 0. At present, only the log2 (0) test is controversial.

Reply:
What do you mean they don't support parsing?

Reply:
Log2(+Infinity) and log2(-Infinity) will report errors in mysql and cannot display the results. Calcite will also report errors in runtime.